src/Components/ClustersListTable/ClustersListTable.spec.ct.js

```diff
@@ -43,21 +156,79 @@ describe('clusters list table', () => {
   });
 
   it('renders table', () => {
-    cy.get(CLUSTERS_LIST_TABLE).should('have.length', 1);
+    cy.get(TABLE).within(() => {
+      cy.get(TOOLBAR).should('have.length', 1);
+      cy.get('table').should('have.length', 1);
+    });
   });
 
-  it('shows 29 clusters as a total number', () => {
-    cy.getTotalClusters().should('have.text', 29);
+  it('shows first clusters only', () => {
+    checkRowCounts(DEFAULT_ROW_COUNT);
+  });
+
+  it('shows correct total number of clusters', () => {
+    checkPaginationTotal(data.length);
+  });
+
+  it('shows table headers', () => {
+    cy.get('table th')
+      .then(($els) => {
+        return _.map(Cypress.$.makeArray($els), 'innerText');
+      })
+      .should('deep.equal', TABLE_HEADERS);
   });
 
-  it('pagination default is set to 20', () => {
-    cy.get(TBODY).children().should('have.length', 20);
+  it(`pagination default is set to ${DEFAULT_ROW_COUNT}`, () => {
     cy.get('.pf-c-options-menu__toggle-text')
       .find('b')
       .eq(0)
       .should('have.text', '1 - 20');
   });
 
+  it('can change page limit', () => {
+    // FIXME: best way to make the loop
+    cy.wrap(PAGINATION_VALUES).each((el) => {
+      cy.get(TOOLBAR).find(PAGINATION_MENU).find(DROPDOWN_TOGGLE).click();
+      cy.get(TOOLBAR)
+        .find(PAGINATION_MENU)
+        .find('ul[class=pf-c-options-menu__menu]')
+        .find(DROPDOWN_ITEM)
+        .contains(`${el}`)
+        .click({ force: true }); // caused by the css issue
```

Comment:
wow, I am surpried we still have the issue with the elements overlapping. must be addressed later on

Reply:
Actually is not needed. I check and most of them can be removed, so I have done